src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,149 @@
 import React from 'react';
+import classNames from 'classnames';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+interface Props {
+  todos: Todo[];
+  onUserSelection: (userId: number) => void;
+}
+
+interface State {
+  title: string;
+  activities: string;
+}
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    title: '',
+    activities: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({ title: value });
+  };
+
+  filterByTitle = () => {
+    const { title } = this.state;
+    const { todos } = this.props;
+
+    const lowerTitle = title.toLowerCase();
+
+    return todos.filter(item => (
+      item.title.includes(lowerTitle)
+    ));
+  };
+
+  selectOption = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    this.setState({ activities: value });
+  };
+
+  filterBySelect = () => {
+    const { activities } = this.state;
+
+    if (activities === 'active') {
+      return (
+        this.filterByTitle().filter(complite => (
+          complite.completed !== true
+        ))
+      );
+    }
+
+    if (activities === 'completed') {
+      return (
+        this.filterByTitle().filter(complite => (
+          complite.completed === true
+        ))
+      );
+    }
+
+    return (
+      this.filterByTitle()
+    );
```

Comment:
```suggestion
    return this.filterByTitle()
```
and better use switch/case